libs/providers/multi-provider/src/lib/errors.ts

```diff
@@ -0,0 +1,53 @@
+import { ErrorCode, GeneralError, OpenFeatureError } from '@openfeature/server-sdk';
+import { PromiseAllSettledResult } from '@opentelemetry/sdk-metrics/build/esnext/utils';
```

Comment:
@ajwootto I think we should not import this here.
We can use the original type here `PromiseSettledResult<Awaited<unknown>>[]` without any import.

Reply:
ah yes the perils of Webstorm auto-import